core/domain/state_domain.py

```diff
@@ -2621,16 +2670,23 @@ def traverse_by_schema_and_convert(
             schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_UNICODE)
 
         if is_subtitled_html_spec or is_subtitled_unicode_spec:
-            value = conversion_fn(value, schema['obj_type'])
+            # Here we use MyPy ignore because here we are assigning
```

Comment:
Wouldn't assertions be more appropriate here (and below)?

Reply:
> Wouldn't assertions be more appropriate here

Assertion won't help, because conversion_fn can take `SubtitledHtml` and `SubtitledUnicode` but one at a time. So if we use assertion here then value will become Union of [ `SubtitledHtml` and `SubtitledUnicode`] and this is not expected by MyPy. 

Another thing is that we are assigning value to the Generic type variable `value`,  which is again not allowed.

Reply:
I see, ok. Yeah, it's not quite the right representation but I agree here that the logical conditions are a bit too complicated here for mypy checking. Thanks for explaining.